### PR TITLE
Add CUTTER_DEPS_DIR option

### LINKED ISSUE #1305: Make build Cutter on Windows easier
**Describe the solution you'd like**

I'd like to build Cutter with python plugins support by issuing one-two commands in console like `prepare_r2.bat && build.bat`

**Describe alternatives you've considered**

Currently I have to prepare full line of a command like this, not convenient:

image

### PR DESCRIPTION
**Pull request by pelijah**
**Detailed description**

**Test plan (required)**

**Closing issues**

Closes #1305

### CODE REVIEW

File: src/Cutter.pro [hunk 1]
```diff
@@ -71,6 +72,15 @@ CUTTER_ENABLE_PYTHON_BINDINGS {
     message("Python Bindings disabled. (requires CUTTER_ENABLE_PYTHON=true)")
 }
 
+defined(CUTTER_DEPS_DIR, var) {
+    !defined(SHIBOKEN_EXECUTABLE, var)          SHIBOKEN_EXECUTABLE="$${CUTTER_DEPS_DIR}\pyside\bin\shiboken2.exe"
+    !defined(SHIBOKEN_INCLUDEDIR, var)          SHIBOKEN_INCLUDEDIR="$${CUTTER_DEPS_DIR}/pyside/include/shiboken2"
```
Comment (xarkes) [inline src/Cutter.pro]: Please choose between slashes or backslashes but don't use both.

File: .appveyor.yml [hunk 2]
```diff
@@ -49,7 +49,7 @@ before_build:
 
 # Build config
 build_script:
-  - cmd: if defined QMAKE ( call prepare_r2.bat && call build.bat CUTTER_APPVEYOR_R2DEC=true CUTTER_ENABLE_CRASH_REPORTS=true CUTTER_ENABLE_PYTHON=true CUTTER_ENABLE_PYTHON_BINDINGS=true SHIBOKEN_EXECUTABLE="%CUTTER_DEPS_DIR%\pyside\bin\shiboken2.exe" SHIBOKEN_INCLUDEDIR="%CUTTER_DEPS_DIR%/pyside/include/shiboken2" SHIBOKEN_LIBRARY="%CUTTER_DEPS_DIR%/pyside/lib/shiboken2.lib" PYSIDE_INCLUDEDIR="%CUTTER_DEPS_DIR%/pyside/include/PySide2" PYSIDE_LIBRARY="%CUTTER_DEPS_DIR%/pyside/lib/pyside2.lib" PYSIDE_TYPESYSTEMS="%CUTTER_DEPS_DIR%/pyside/share/PySide2/typesystems")
+  - cmd: if defined QMAKE ( call prepare_r2.bat && call build.bat CUTTER_APPVEYOR_R2DEC=true CUTTER_ENABLE_CRASH_REPORTS=true CUTTER_ENABLE_PYTHON=true CUTTER_ENABLE_PYTHON_BINDINGS=true CUTTER_DEPS_DIR="%CUTTER_DEPS_DIR%)
```
Comment (a1ext) [inline .appveyor.yml]: did you forget closing `"` ? Appveyor is red

**Comment by a1ext** [comment]
> I'm not really found of this approach because the user will have to have exactly the same directory structure and cannot have another name for the pyside folder (e.g. `pyside_v1.xx_win64` or I don't know).
> Thoughts @a1ext?

He just refactored the code which we already had before. User had no choice with pyside folder naming before as well.

File: src/Cutter.pro (hunk 1 above)
Comment (thestr4ng3r) [reply src/Cutter.pro]: Always use slashes

**Comment by pelijah** [comment]
> 
> 
> I'm not really found of this approach because the user will have to have exactly the same directory structure and cannot have another name for the pyside folder (e.g. `pyside_v1.xx_win64` or I don't know).

Right, because it is assumed that user download deps from https://github.com/radareorg/cutter-deps. But user still is able to override some paths. 
In general I don't see other way to ease the build. We added these options to provide users a way to build Cutter without extra dependencies. What do you propose? Make these dependency mandatory and add them to the codebase?

**Comment by xarkes** [comment]
Well, right now I don't propose anything.. Adding the dependencies to the codebase is not a good idea, so let's stick with what we have :)

**Pull request merged**